### PR TITLE
Chat commands

### PR DESCRIPTION
**Pull request by Freddo3000**
This pull request will add a bunch of chat commands for various utility purposes. Each of them is able to be disabled via CBA options.  
__Available options are:__
* Always available
* Available during safestart
* Available during safestart & after respawn
* Never available

Admins have access to these commands at all times.

----
### #heal
Description: Heals player/target player

Syntaxes:  
#heal - Heals self  
#heal \<player> - Heals target player  
#heal all - Heals everyone

----
### #radio
Description: Adds a radio to player

Syntaxes:  
#radio \<radioType> - Adds defined radio to player. Displayname and classnames are checked

----
### #rp
Description: Resets player to a nearby empty position. Useful for getting yourself unstuck from a rock.

Syntaxes:  
#rp - Resets player position

----
### #spec
Description: Puts you into spectator.

Syntaxes:
#spec - Puts player into spectator.

----
### #stage
Description: Teleports player group to target position, used for staging before mission.  
Also has new module associated to it: TMF_chat_adversarialSafeArea. Limits teleport to areas outside synchronized TMF areas, dependending on synchronized side logics.

Syntaxes:  
#stage - Only available to group leader. Opens map and allows you to select a position to teleport to.

----
### #tp

Description: Teleports to target player/group

Syntaxes:   
#tp \<player>|\<group> - Teleports to target player, or group leader.

----
### #whisper
Description: Sends a whisper to target player in hint format.

Syntaxes:
#whisper \<player> \<message> - Sends message to target player.

-----

Credits to @jameslkingsley for the initial implementation in [ARCORE](https://github.com/ARCOMM/ARCORE)

### CODE REVIEW

**Comment by Freddo3000** [comment]
Merge this before #285

**Comment by headswe** [comment]
I don't really see the need for the chat commands to be part of the core TMF.

**Comment by Freddo3000** [comment]
It is very useful to have during impromptu missions, like Zeus Game Master and similar where the entire template isn't set up. Staging to positions, manually entering spectator and other functions that would otherwise have to be set up as modules pre-mission, or someone to use the admin menu for.

Having the whisper command also helps declutter the chat in case of issues, and removes the necessity of having the session host on your friends list.

I think they are suitable to have in the framework, and as mentioned they can be disabled through CBA options if you'd rather not have them. Infact the default setting is for them to be disabled.

**Comment by Sniperhid** [comment]
So default is off which we like which gives communities the option to enable it should they want it :).

The only command we have an issue with currently is `#stage` as this should be integrated into the teleport module (so using the area module should also effect teleport etc). What I suggest is we break that out into a separate PR.

File: addons/chat/functions/fn_cmndRP.sqf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Description:
+ * See Syntaxes
+ */
+
+#include "\x\tmf\addons\chat\script_component.hpp"
+
+IS_CMND_AVAILABLE(GVAR(rpUsage),"#rp");
+
+private _safePos = (getPosATL player) findEmptyPosition [0, 25, (typeOf player)];
```
Comment (Sniperhid) [inline addons/chat/functions/fn_cmndRP.sqf]: suggest using `[] call CBA_fnc_currentUnit` to get player as would handle case of zeus usage.

**Pull request merged**